### PR TITLE
ca-list ported

### PR DESCRIPTION
**Pull request by tapina**
Tests pass and example works.

### CODE REVIEW

File: components/ca-list.css [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    position: absolute;
+    right: 2px;
+    bottom: -22px;
+    font-size: 0.9em;
+    color: #888;
+}
+
+ca-list table[data-count="1"]:after {
+    content: attr(data-count) " item in total";
+}
```
Comment (nealeu) [inline components/ca-list.css]: Wow.. use of a feature that's been available for 7 years (in IE9), but took the death of IE8 to be able to actually use the thing!

File: test/ca-list-test.html [hunk 2]
```diff
@@ -0,0 +1,497 @@
+<!doctype html>
+<html>
+<head>
+    <meta charset="utf-8">
+    <title>CA FORM | Web Lab Common UI</title>
```
Comment (nealeu) [inline test/ca-list-test.html]: CA LIST

File: components/ca-list.css [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
...vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCIgc3Ryb2tlPSJub25lIiBzdHJva2Utd2lkdGg9IjEiPiAgICA8ZyB0cmFuc2Zvcm09InRyYW5zbGF0ZSgtNTM5LjAwMDAwMCwgLTQwNy4wMDAwMDApIj4gICAgICAgIDxnIHRyYW5zZm9ybT0idHJhbnNsYXRlKDIxNS4wMDAwMDAsIDExOS4wMDAwMDApIi8+ICAgICAgICA8cGF0aCBkPSJNNTU1LjQ3NzI3Niw0MjEuMzU1OTU2IEM1NTYuNDM3MTIsNDE5Ljk3OTM4MyA1NTcsNDE4LjMwNTQyNSA1NTcsNDE2LjUgQzU1Nyw0MTEuODA1NTc5IDU1My4xOTQ0MjEsNDA4IDU0OC41LDQwOCBDNTQzLjgwNTU3OSw0MDggNTQwLDQxMS44MDU1NzkgNTQwLDQxNi41IEM1NDAsNDIxLjE5NDQyMSA1NDMuODA1NTc5LDQyNSA1NDguNSw0MjUgQzU1MC4zMDUxNDgsNDI1IDU1MS45Nzg4NjgsNDI0LjQzNzI5MyA1NTMuMzU1MzIxLDQyMy40Nzc3MTkgTDU1My4zNTU4OTIsNDIzLjQ3NzIxMiBMNTU5LjY1OTQzMyw0MjkuNzgwNzUzIEM1NTkuNzc2MzA5LDQyOS44OTc2MjkgNTU5Ljk2MjIwNiw0MjkuOTAxMjI1IDU2MC4wODIyMTEsNDI5Ljc4MTIyIEw1NjEuNzgxMjIsNDI4LjA4MjIxMSBDNTYxLjg5NzgzOCw0MjcuOTY1NTkzIDU2MS44OTI0MTcsNDI3Ljc3MTA5NyA1NjEuNzgwNzUzLDQyNy42NTk0MzMgTDU1NS40NzcyNzYsNDIxLjM1NTk1NiBaIE01NDguNSw0MjMgQzU1Mi4wODk4NTEsNDIzIDU1NSw0MjAuMDg5ODUxIDU1NSw0MTYuNSBDNTU1LDQxMi45MTAxNDkgNTUyLjA4OTg1MSw0MTAgNTQ4LjUsNDEwIEM1NDQuOTEwMTQ5LDQxMCA1NDIsNDEyLjkxMDE0OSA1NDIsNDE2LjUgQzU0Miw0MjAuMDg5ODUxIDU0NC45MTAxNDksNDIzIDU0OC41LDQyMyBaIiBmaWxsPSIjMzMzIi8+ICAgICAgICA8L2c+ICAgIDwvZz48L3N2Zz4=) no-repeat 98% center;
```
Comment (norisalsaadie) [inline components/ca-list.css]: Not related to your changes, but I just noticed this. This sort of thing should really be left for a build step...

File: components/ca-list.js [hunk 4]
```diff
@@ -0,0 +1,722 @@
...
+            if (this.sortIndex > -1) this.sortTable();
+        }
+
+        // noinspection JSUnusedGlobalSymbols
+        /**
+         * Executes when the element is attached to the DOM
+         * @returns {undefined} nothing
+         */
+        attachedCallback() {
+            this.attached = true;
```
Comment (norisalsaadie) [inline components/ca-list.js]: Not related to your changes, but is this state really needed? 
If you have a parent node and you are a root custom element, then you are attached.

File: components/ca-list.js [hunk 5]
```diff
@@ -0,0 +1,722 @@
...
+
+                    case 'src':
+                        this.loadSrc();
+                        break;
+
+                    case 'sort-index':
+                        this.sortTable();
+                        break;
+
+                    default:
```
Comment (norisalsaadie) [inline components/ca-list.js]: Nice addition, the previous file didn't have a default 👍

File: components/ca-list.js [hunk 6]
```diff
@@ -0,0 +1,722 @@
...
+            }
+        }
+
+        /** @return {string} url to schema or instance data containing HATEOAS links */
+        get src() {
+            return this.getAttribute('src') || '';
+        }
+
+        /** @param {string} value - url to schema or instance data containing HATEOAS links */
+        set src(value) {
```
Comment (norisalsaadie) [inline components/ca-list.js]: I know the old code doesn't have default params, but shouldn't we use this where non-breaking changes are obvious?
`set src(value = '')`

File: test/ca-list-test.html [hunk 7]
```diff
@@ -0,0 +1,497 @@
...
+            caList.data = testData.caListData.list;
+
+            // get all rendered table rows
+            const rows = Array.prototype.slice.call(caList.querySelectorAll('table tbody tr'));
+
+            // go through each row in the table
+            rows.forEach(function (row, rowIndex) {
+
+                const dataRow = testData.caListData.list[rowIndex], // get related data object for this row
+                        cols = Array.prototype.slice.call(row.cells); // go columns in this table row as an array
```
Comment (norisalsaadie) [inline test/ca-list-test.html]: does this pass the airbnb linting rules?

**Pull request merged**